options/ansi/generic/file-io.cpp

```diff
@@ -237,9 +237,13 @@ void abstract_file::purge() {
 	__dirty_end = __dirty_begin;
 }
 
-// TODO: For input files, discard the buffer.
 int abstract_file::flush() {
-	return _write_back();
+	if (__dirty_end == __dirty_begin) {
+		return _discard_buffer;
```

Comment:
Why don't we discard the buffer unconditionally?

Reply:
Spec states that we should only discard the buffer for input streams, and output streams should write back to the FD

Reply:
I think here we should really make sure that we do the correct thing for r/w files. Can we maybe write a test case to validate that our behavior matches glibc/musl?